src/UserAction/ConfirmationExecutor.php

```diff
@@ -68,7 +68,7 @@ public function afterActionInit(Model\UserAction $action)
 
         $table_name = is_array($action->getModel()->table) ? $getTableName($action->getModel()->table) : $action->getModel()->table;
 
-        $this->id = mb_strtolower($this->name . '_' . $table_name . '_' . $action->short_name);
+        $this->id = preg_replace('~[^A-Za-z0-9\-]~', '_', mb_strtolower($this->name . '_' . $table_name . '_' . $action->short_name));
```

Comment:
`extends Modal implements JsExecutorInterface` ... the same base

can we dedup these classes a little bit?